de-lib/src/main/java/org/iplantc/de/client/services/impl/DiskResourceServiceFacadeImpl.java

```diff
@@ -969,11 +969,31 @@ protected MetadataTemplate convertFrom(String object) {
     }
 
     @Override
-    public void shareWithAnonymous(HasPaths diskResourcePaths, DECallback<String> callback) {
+    public void shareWithAnonymous(final HasPaths diskResourcePaths, DECallback<List<String>> callback) {
         String address = deProperties.getDataMgmtBaseUrl() + "anon-files"; //$NON-NLS-1$
         final String body = encode(diskResourcePaths);
         ServiceCallWrapper wrapper = new ServiceCallWrapper(POST, address, body);
-        callService(wrapper, callback);
+
+        callService(wrapper, new DECallbackConverter<String, List<String>>(callback) {
+            @Override
+            protected List<String> convertFrom(final String json) {
+                Splittable obj = StringQuoter.split(json);
```

Comment:
I'm pretty sure this is a special case, since the endpoint returns a response like the following:
```json
{
  "/path/1": "http://url/1",
  "/path/2": "http://url/2"
}
```
I'm not sure if AutoBeans can handle variable keys like this, and since we only need the URL values, it seemed easier to parse them into a list, while also keeping this parsing logic isolated in the service facade.